AU2/plugins/custom_plugins/CompetencyPlugin.py

```diff
@@ -71,6 +74,84 @@
     INCOS_PAGE_TEMPLATE = F.read()
 
 
+class PlayerStatus(enum.Enum):
+    COMPETENT = enum.auto()
+    POLICE = enum.auto()
+    INCOMPETENT = enum.auto()
+    GIGAINCOMPETENT = enum.auto()
+    DEAD = enum.auto()
+
+    def is_alive_player(self):
+        """
+        Returns true if this player status represents a player that is alive
+        """
+        return self in [self.COMPETENT, self.INCOMPETENT, self.GIGAINCOMPETENT]
+
+    def __str__(self):
+        return str(self.name).replace("PlayerStatus.", "")
+
+
+@dataclasses.dataclass
+class PlayerInfo:
+    identifier: str
+    status: PlayerStatus
+    competency_deadline: datetime.datetime
+
+
+def get_active_players(death_manager: DeathManager) -> Set[str]:
+    """
+    Collects all players not currently at risk of a gigabolt
+    """
+    active_players = []
+
+    for e in sorted(list(EVENTS_DATABASE.events.values()), key=lambda event: event.datetime):
+        death_manager.add_event(e)
+        for killer, _ in e.kills:
+            active_players.append(killer)
+        for player_id in e.pluginState.get("CompetencyPlugin", {}).get("attempts", []):
+            active_players.append(player_id)
+    return set(active_players)
+
+
+def get_player_infos(from_date=get_now_dt()) -> Dict[str, PlayerInfo]:
+    """
+    Returns a list of calculated player informations (see struct above)
+    """
+    events = list(EVENTS_DATABASE.events.values())
+    events.sort(key=lambda event: event.datetime)
+    start_datetime: datetime.datetime = get_game_start()
+
+    competency_manager = CompetencyManager(start_datetime)
+    death_manager = DeathManager(perma_death=True)
+
+    # populates the death manager
+    active_players = get_active_players(death_manager)
+
+    for e in events:
+        competency_manager.add_event(e)
+
+    infos = {}
+    for a in ASSASSINS_DATABASE.get_filtered(include_hidden=lambda _: True):
+        status = PlayerStatus.COMPETENT
+        is_gigainco = a.identifier not in active_players
+        is_inco = competency_manager.is_inco_at(a, from_date)
+        is_dead = death_manager.is_dead(a)
+        if a.is_police:
+            status = PlayerStatus.POLICE
+        elif is_gigainco and not is_dead:
+            status = PlayerStatus.GIGAINCOMPETENT
+        elif is_inco and is_dead:
+            status = PlayerStatus.DEAD

```

Comment:
Actually, on second thoughts I'm not sure I agree with the logic here, it seems misleading for dead players who are still within their competency deadlines not to be marked as "dead".